### PR TITLE
Task#173629 fix: Asset Library Integration

### PR DESCRIPTION
**Pull request by AnkitaMaheshChavan**

### CODE REVIEW

File: projects/event-library/src/lib/events/components/asset-browser/asset-browser.component.html [hunk 1]
```diff
@@ -0,0 +1,127 @@
+<sui-modal [isClosable]="true" class="sb-modal overflow-modal" [isInverted]="false" (dismissed)="dismissImagePicker()"
+  *ngIf="showImagePicker" [size]="'normal'" [isFullScreen]="false" [mustScroll]="true" #modal>
+  <div class="sb-modal-header">
+    Select Image
+  </div>
+  <div class="sb-modal-content p-0">
+    <sui-tabset>
+      <div class="sb-tabset-menu">
+        <a class="sb-item" (activate)="getMyImages(0)" suiTabHeader="1">My Images</a>
```
Comment (ahghatol) [inline projects/event-library/src/lib/events/components/asset-browser/asset-browser.component.html:9]: Remove the hardcoded text messages, add it to lables.ts and use it.

File: projects/event-library/src/lib/events/components/asset-browser/asset-browser.component.css [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+    padding: 0px;
+    cursor: pointer;
+    opacity: 0;
+    height: 100%;
+} 
+}*/
+
+.upload-file-section {
+  display: flex;
+  height: 240px;
```
Comment (ahghatol) [inline projects/event-library/src/lib/events/components/asset-browser/asset-browser.component.css:197]: Get the CSS review from Mangesh

File: projects/event-library/src/lib/events/components/asset-browser/asset-browser.component.ts [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+import {config} from './asset-browser.data';
+import * as _ from 'lodash-es';
+
+@Component({
+  selector: 'lib-asset-browser',
+  templateUrl: './asset-browser.component.html',
+  styleUrls: ['./asset-browser.component.css']
+})
+export class AssetBrowserComponent implements OnInit, OnDestroy {
+ // @ViewChild('modal') modal;
```
Comment (ahghatol) [inline projects/event-library/src/lib/events/components/asset-browser/asset-browser.component.ts:13]: @AnkitaMaheshChavan Remove the unsed code

File: projects/event-library/src/lib/events/components/asset-browser/asset-browser.component.ts [hunk 4]
```diff
@@ -0,0 +1,308 @@
...
+  }
+  ngOnDestroy() {
+    // if (this.modal && this.modal.deny) {
+    //   this.modal.deny();
+    // }
+  }
+
+  // Function to get new images
+  // getmyImages()
+  // {
```
Comment (ahghatol) [inline projects/event-library/src/lib/events/components/asset-browser/asset-browser.component.ts:63]: @AnkitaMaheshChavan  make sure you clean all the unused code

File: projects/event-library/src/lib/events/components/events.module.ts [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+import { CoverEventDetailComponent } from '../components/cover-event-detail/cover-event-detail.component';
+// import { EventFilterComponent} from './events/components/event-filter/event-filter.component';
+import { EventFilterComponent} from '../components/event-filter/event-filter.component';
+import { EventRoutingModule } from '../event-routing.module';
+import { FormsModule, ReactiveFormsModule } from '@angular/forms';
+import { CommonFormElementsModule } from 'common-form-elements';
+import { NgxIziToastModule } from 'ngx-izitoast';
+import { TranslateModule, TranslateLoader } from '@ngx-translate/core';
+import { TranslateHttpLoader } from '@ngx-translate/http-loader';
+import { HttpClientModule, HttpClient } from '@angular/common/http';
```
Comment (ahghatol) [inline projects/event-library/src/lib/events/components/events.module.ts:17]: @AnkitaMaheshChavan This file is already in lib, why do we need it here?

File: projects/event-library/src/lib/events/components/event-create/event-create.component.ts [hunk 6]
```diff
@@ -189,5 +209,36 @@ export class EventCreateComponent implements OnInit {
...
+    
+    this.appIcon="";
+    if (this.isReviewMode()) {
+      this.appIconConfig = {...this.appIconConfig , ... {isAppIconEditable: false}};
+    } else {
+      this.appIconConfig = {...this.appIconConfig , ... {isAppIconEditable: true}};
+    }
+  }
+
+  isReviewMode()
```
Comment (ahghatol) [inline projects/event-library/src/lib/events/components/event-create/event-create.component.ts:230]: We don't have review mode for the event

File: projects/event-library/src/lib/events/components/event-create/event-create.component.ts [hunk 7]
```diff
@@ -189,5 +209,36 @@ export class EventCreateComponent implements OnInit {
...
+  {
+    this.imageSearchService.getEditMode().subscribe((data: any) => {
+    this.editmode = data.d.edit;
```
Comment (ahghatol) [inline projects/event-library/src/lib/events/components/event-create/event-create.component.ts:233]: @AnkitaMaheshChavan Recheck this code and keep only what we need for event lib

File: projects/event-library/src/lib/events/components/collection-icon/collection-icon.component.ts [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+
+@Component({
+  selector: 'lib-collection-icon',
+  templateUrl: './collection-icon.component.html',
+  styleUrls: ['./collection-icon.component.scss']
+})
+export class CollectionIconComponent implements OnInit {
+  //public showImagePicker : boolean;
+  public showImagePicker = false;
+  @Output() iconEmitter = new EventEmitter<any>();
```
Comment (ahghatol) [inline projects/event-library/src/lib/events/components/collection-icon/collection-icon.component.ts:12]: Rename this component to EventIconComponent

File: projects/event-library/src/lib/events/components/asset-browser/asset-browser.component.css [hunk 9]
```diff
@@ -0,0 +1,283 @@
...
+  flex-direction: column;
+  margin: 0 auto;
+  background-color: #F5F9FC;
+  border: 1px dashed #80a7ce;
+}
+
+.qq-uploader.qq-uploader-selector.custom-qq-uploader {
+  background: inherit;
+  border-color: none;
+  border: none;
```
Comment (ahghatol) [inline projects/event-library/src/lib/events/components/asset-browser/asset-browser.component.css:211]: @AnkitaMaheshChavan  Hope you have kept only the CSS needed for asset lib

File: projects/event-library/src/lib/events/events.module.ts [hunk 10]
```diff
@@ -15,6 +15,11 @@ import { TranslateModule, TranslateLoader } from '@ngx-translate/core';
 import { TranslateHttpLoader } from '@ngx-translate/http-loader';
 import { HttpClientModule, HttpClient } from '@angular/common/http';
 import {NgxPaginationModule} from 'ngx-pagination';
+import { CollectionIconComponent } from './components/collection-icon/collection-icon.component';
+import { AssetBrowserComponent } from './components/asset-browser/asset-browser.component'; // <-- import the module
+import { SuiModule } from 'ng2-semantic-ui/dist';
+import { InfiniteScrollModule } from 'ngx-infinite-scroll';
```
Comment (ahghatol) [inline projects/event-library/src/lib/events/events.module.ts:21]: "ngx-infinite-scroll": "^6.0.0",

@AnkitaMaheshChavan Add this module in your pacakge.json dependencies too

File: projects/event-library/src/lib/events/events.module.ts [hunk 11]
```diff
@@ -15,6 +15,11 @@ import { TranslateModule, TranslateLoader } from '@ngx-translate/core';
 import { TranslateHttpLoader } from '@ngx-translate/http-loader';
 import { HttpClientModule, HttpClient } from '@angular/common/http';
 import {NgxPaginationModule} from 'ngx-pagination';
+import { CollectionIconComponent } from './components/collection-icon/collection-icon.component';
+import { AssetBrowserComponent } from './components/asset-browser/asset-browser.component'; // <-- import the module
+import { SuiModule } from 'ng2-semantic-ui/dist';
```
Comment (ahghatol) [inline projects/event-library/src/lib/events/events.module.ts:20]: @AnkitaMaheshChavan  Recheck this(SuiModule) with Mohini P as she is working on upgrading this module to angular version 9 module, also make sure you add it in package.json

File: projects/event-library/src/lib/events/services/image-search/image-search.service.ts [hunk 12]
```diff
@@ -0,0 +1,77 @@
...
+          request: {
+            filters: {
+              contentType: 'Asset',
+              compatibilityLevel: {
+                min: 1,
+                max: 2
+              },
+              status: ['Live'],
+            },
+            limit: 50,
```
Comment (ahghatol) [inline projects/event-library/src/lib/events/services/image-search/image-search.service.ts:42]: @AnkitaMaheshChavan  pass limit as an optional parameter to this function.

File: src/assets/api/editmode.json [hunk 13]
```diff
@@ -0,0 +1,8 @@
+{
+    "d":{
+    "edit": "edit",
+    "review": "review",
```
Comment (ahghatol) [inline src/assets/api/editmode.json:4]: @AnkitaMaheshChavan  This is specific to collection editor, Do we need this in event lib ?

File: src/assets/api/search-images.json [hunk 14]
```diff
@@ -0,0 +1,2758 @@
...
+          "artifactUrl": "https://sunbirddev.blob.core.windows.net/sunbird-content-dev/content/do_11281332607717376012/artifact/033019_sz_reviews_feat_1564126718632.jpg",
+          "contentEncoding": "identity",
+          "lastUpdatedOn": "2019-07-26T07:38:39.907+0000",
+          "contentType": "Asset",
+          "dialcodeRequired": "No",
+          "identifier": "do_11281332607717376012",
+          "creator": "Creation ",
+          "audience": [
+            "Learner"
+          ],
```
Comment (ahghatol) [inline src/assets/api/search-images.json:102]: @AnkitaMaheshChavan Make sure you remove this file once we have API integation

File: projects/event-library/src/lib/events/components/asset-browser/asset-browser.component.html (hunk 1 above)
Comment (AnkitaMaheshChavan) [reply projects/event-library/src/lib/events/components/asset-browser/asset-browser.component.html:9]: resolved

File: src/assets/api/editmode.json (hunk 13 above)
Comment (AnkitaMaheshChavan) [reply src/assets/api/editmode.json:4]: yes sir. Based on this function calls are dependent.

**Pull request merged**